filebeat/beater/filebeat.go

```diff
@@ -123,15 +123,15 @@ func newBeater(b *beat.Beat, plugins PluginFactory, rawConfig *conf.C) (beat.Bea
 	}
 
 	if b.API != nil {
-		if err = inputmon.AttachHandler(b.API.Router()); err != nil {
+		if err = inputmon.AttachHandler(b.API.Router(), b.Info.Monitoring.NamespaceRegistry()); err != nil {
```

Comment:
I think I'm ready to approve except for one last detail, I was still hoping we could have the inputs below a top-level key (probably "inputs") in the namespace registry instead of just at the root, since as far as I know there's nothing stopping a user from giving an input an id that collides with one of our own internal registry trees like "stats". (Unless that would break compatibility in some way I'm not aware of.) If so it might make sense to make an `InputMetricsRegistry()` helper on `Monitoring` and call that instead of `NamespaceRegistry()` here and elsewhere.

Reply:
I thought about it too. However currently we have the backwards compatibility issue: the legacy API has to access use the same registry as the new API. Otherwise we'd end up with the current metrics on a registry and the pipeline metrics on another.

My idea has been to create the input registry and its access method,`Monitoring.InputMetricsRegistry()`, once, or at the same time, we get rid of the legacy API, at least for filebeat.

What do you think?

That's also why I didn't created a `InputMetricsRegistry()`. It'd just be returning the namespace registry, but would it seem like it's a input specific registry as it's for the others. I rather not have a misleading API, so I kept it out for now.

Reply:
I started digging into this and ended up spending all day untangling things... the short version is, the current use of `Info.Monitoring.Namespace` is already "broken" in ways that can't be fixed in the scope of this PR, so I think I agree, let's merge it this way since it looks the same to the individual Beats, and I'll do some urgent followup work to get the metrics registry hierarchies behaving more consistently between classic beats and OTel before the ambiguity spreads any further :-)